indigo_app/static/javascript/indigo/models.js

```diff
@@ -40,27 +45,86 @@
       return this.document.url() + '/content';
     },
 
+    setupMutationObserver: function () {
+      this.observer = new MutationObserver((mutations) => {
+        for (const mutation of mutations) {
+          console.log('mutation', mutation);
+          this.trigger('mutation', this, mutation);
+        }
+        this.trigger('change:dom', this);
+        this.trigger('change', this);
+      });
+
+      this.observer.observe(this.xmlDocument, {
+        childList: true,
+        attributes: true,
+        subtree: true,
+      });
+    },
+
+    /**
+     * Determine the impact of a mutation on the provided element.
+     * @param mutation MutationRecord
+     * @param element Element in this XML document
+     * @returns 'changed' if the mutation impacts the element, 'removed' if the element was removed from the tree,
+     *          'replaced', if the element has been replaced (with a node in mutation.addedNodes),
+     *          or null if there is no impact
+     */
+    getMutationImpact(mutation, element) {
+      const target = mutation.target;
+
+      if (mutation.type === 'childList') {
+        if (mutation.removedNodes[0] === element) {
+          // the element has been replaced
+          return 'replaced';
+        }
+
+        const ownerDocument = target.nodeType === Node.DOCUMENT_NODE ? target : target.ownerDocument;
+        if (!ownerDocument.contains(element)) {
+          // the change removed xmlElement from the tree
+          return 'removed';
+        }
+      }
+
+      if (target === element || element.contains(target)) {
+        // the mutated node is xmlElement itself or one of its descendants
+        return 'changed';
+      }
+
+      // we don't care about attribute or character data changes elsewhere in the document
+    },
+
     contentChanged: function(model, newValue, options) {
-      // don't bother updating the DOM if the source of this event
-      // is already a change to the DOM
-      if (options && options.fromXmlDocument) return;
+      let root = null;
```

Comment:
initially setting the content triggers this, so the Save button is enabled from the start

Reply:
But changing `this.documentContent.trigger('sync');` to `this.document.trigger('sync');` at 
https://github.com/laws-africa/indigo/pull/2320/files#diff-f541713a19b39bec6d0c84799a3285e34a13cf02ec2e2a471ba9d7973ddd80a7L220 
fixes it (I can't comment there 🤷)

Reply:
fixed